File: android/src/main/kotlin/com/nwdn/baji/FlutterBajiPlugin.kt
package com.nwdn.baji

import android.content.Context
import android.graphics.BitmapFactory
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result


/** FlutterBajiPlugin */
class FlutterBajiPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel
    private lateinit var context: Context  // 添加 context 变量

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        context = flutterPluginBinding.applicationContext  // 保存 context
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flutter_baji")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        if (call.method == "getPlatformVersion") {
            result.success("Android ${android.os.Build.VERSION.RELEASE}")
        } else if (call.method == "saveImageWithDpi") {
            val imagePath = call.argument<String>("imagePath")
            val dpi = call.argument<Int>("dpi") ?: 300
            val name = call.argument<String>("name")
            val isJpg = call.argument<Boolean>("isJpg") ?: true

            if (imagePath == null || name == null) {
                result.error("INVALID_ARGUMENTS", "Missing required arguments", null)
                return
            }

            try {
                val bitmap = BitmapFactory.decodeFile(imagePath)
                val savedFile = ImageDpiUtil.saveBitmapToJpgWithDpi(
                    context,
                    bitmap,
                    dpi,
                    name,
                    isJpg
                )
                result.success(savedFile?.absolutePath)
            } catch (e: Exception) {
                result.error("SAVE_ERROR", e.message, null)
            }
        } else {
            result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}

File: ios/Classes/FlutterBajiPlugin.swift
import Flutter
import UIKit

public class FlutterBajiPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "flutter_baji", binaryMessenger: registrar.messenger())
        let instance = FlutterBajiPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }
    
    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getPlatformVersion":
            result("iOS " + UIDevice.current.systemVersion)
        case "saveImageWithDpi":
            if let args = call.arguments as? [String: Any],
               let imagePath = args["imagePath"] as? String {
                
                guard let image = UIImage(contentsOfFile: imagePath),
                      let name = args["name"] as? String else {
                    result(FlutterError(code: "INVALID_IMAGE",
                                      message: "Could not load image",
                                      details: nil))
                    return
                }
                
                let dpi = args["dpi"] as? Int ?? 300
                let isJpg = args["isJpg"] as? Bool ?? true
                if let savedPath = ImageDpiUtil.saveImageWithDpi(image, name: name, dpi: dpi, isJpg: isJpg) {
                    result(savedPath)
                } else {
                    result(FlutterError(code: "SAVE_ERROR",
                                      message: "Failed to save image",
                                      details: nil))
                }
            } else {
                result(FlutterError(code: "INVALID_ARGUMENTS",
                                  message: "Missing required arguments",
                                  details: nil))
            }
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
